docstore/doc.go

```diff
@@ -70,6 +70,8 @@
 //       "PublicationYear": 1967,
 //    }
 //
+// Note that the value of "Author" is itself a document.
```

Comment:
This helps. Another option would be to inline type comment next to the field or even each field for visual labeling. Or you could embed these into field names for self-documenting code.